src/Microsoft.SqlTools.ServiceLayer/LanguageServices/PeekDefinition.cs

```diff
@@ -37,9 +41,21 @@ private Database Database
             {
                 if (this.connectionInfo.SqlConnection != null)
                 {
-                    Server server = new Server(this.connectionInfo.SqlConnection.DataSource);
-                    return server.Databases[this.connectionInfo.SqlConnection.Database];
-
+                    try
+                    {
+                        // Get server object from connection
+                        string connectionString = ConnectionService.BuildConnectionString(this.connectionInfo.ConnectionDetails);
+                        SqlConnection sqlConn = new SqlConnection(connectionString);                    
+                        sqlConn.Open();
+                        ServerConnection serverConn = new ServerConnection(sqlConn); 
+                        Server server = new Server(serverConn);
+                        return server.Databases[this.connectionInfo.SqlConnection.Database];
+                    }
+                    catch(Exception ex)
+                    {
+                        Logger.Write(LogLevel.Error, ex.Message);
```

Comment:
Please help identify the source of the error message - for example "PeekDefinition.Database - Unexpected exception " + ex.Message